### PR TITLE
Br credit cards

### PR DESCRIPTION
**Pull request by fonse**
Adding hypercard/aura/elo cards to recognized cards array.

Note that I had to move hypercard's priority above diners (conflict on `38*`) and discover (conflict on `60*`).

The conflict with discover isn't that big, since the card numbers' lengths are compatible.
However, this is not the case for diners, as seen in the commented test for `38520000023237`.

### CODE REVIEW

**Comment by fonse** [comment]
@lombo Potencialmente rompemos diners con esto. Salimos igual?

BTW, no se si fue a propósito o no, pero el PR que armaste vos es contra el repo original.

**Comment by martinacortes** [comment]
Esto fixea el issue https://github.com/TiendaNube/issues/issues/9290 !

File: src/jquery.payment.coffee [hunk 1]
```diff
@@ -76,6 +76,14 @@ $.payment.cards = cards = [
     luhn: true
   }
   {
+    type: 'hipercard'
```
Comment (lombo) [inline src/jquery.payment.coffee]: Pongamos un comment acá tal cual lo que dice el PR para que quede claro por qué lo pusimos antes y donde conflictea

File: test/specs.coffee [hunk 2]
```diff
@@ -62,12 +62,12 @@ describe 'jquery.payment', ->
...
 
       assert($.payment.validateCardNumber('30569309025904'), 'dinersclub')
-      assert($.payment.validateCardNumber('38520000023237'), 'dinersclub')
+      # assert($.payment.validateCardNumber('38520000023237'), 'dinersclub')
       assert($.payment.validateCardNumber('36700102000000'), 'dinersclub')
       assert($.payment.validateCardNumber('36148900647913'), 'dinersclub')
 
-      assert($.payment.validateCardNumber('6011111111111117'), 'discover')
-      assert($.payment.validateCardNumber('6011000990139424'), 'discover')
+      assert($.payment.validateCardNumber('6011111111111117'), 'discover') # Actually considered hipercard
```
Comment (lombo) [inline test/specs.coffee]: Hay que ponerle hipercard, no?

**Comment by lombo** [comment]
Te dejé unos comments, pero OK!

"Potencialmente rompe diners" AKA la reconoce como hipercard, no?

**Comment by lombo** [comment]
Claro, habría que hacer un manejito adicional de "país" para que solo tome hipercard si es de BR, pero bueno salgamos con esto y después arrumamos

File: src/jquery.payment.coffee (hunk 1 above)
Comment (lombo) [inline src/jquery.payment.coffee:79]: Let's add the PR comment over here so it's really clear why we are putting hipercard over diners!

File: test/specs.coffee (hunk 2 above)
Comment (lombo) [inline test/specs.coffee:69]: Should be hipercard? Is this test passing?

**Comment by lombo** [comment]
Left some comments, but code is :100: 

The diners issue means that on countries other than BR, we would recognize some Diners CC's as Hipercards, right?

**Pull request closed without merging**